python/rmm/rmm/_lib/memory_resource.pyx

```diff
@@ -1324,3 +1330,87 @@ def available_device_memory():
     cdef pair[size_t, size_t] res
     res = c_available_device_memory()
     return (res.first, res.second)
+
+
+cdef dict byte_sizes = {
+    "kB": 10 ** 3,
+    "MB": 10 ** 6,
+    "GB": 10 ** 9,
+    "TB": 10 ** 12,
+    "PB": 10 ** 15,
+    "KiB": 2 ** 10,
+    "MiB": 2 ** 20,
+    "GiB": 2 ** 30,
+    "TiB": 2 ** 40,
+    "PiB": 2 ** 50,
+    "B": 1,
+    "": 1,
+}
+
+byte_sizes.update({k.lower(): v for k, v in byte_sizes.items()})
+byte_sizes.update({k[0]: v for k, v in byte_sizes.items() if k and "i" not in k})
+byte_sizes.update({k[:-1]: v for k, v in byte_sizes.items() if k and "i" in k})
+
+cdef int parse_bytes(object s):
```

Comment:
Note for reviewer: This function was taken from [dask](https://github.com/dask/dask/blob/31af7f7040643c447a72c87a8f12457094ec15ff/dask/utils.py#L1171)

Reply:
What, if any, are the license implications of this appropriation?

Reply:
Not sure, let me ask the Dask team.

Reply:
Dask is licensed under BSD 3-clause. So at the very least, you would need to reproduce the license text.

Reply:
Resolving this since we've changed the approach